devito/types/misc.py

```diff
@@ -111,6 +112,39 @@ def free_symbols(self):
         return (super().free_symbols |
                 set().union(*[i.free_symbols for i in self.strides]))
 
+    def bind(self, pname):
+        """
+        Generate a 2-tuple:
+
+            * A macro which expands to the 1-dimensional representation of the
+              FIndexed, e.g. `aL0(t,x,y) -> a[(t)*x_stride0 + (x)*y_stride0 + (y)]`
+            * A new FIndexed, with the same indices as `self`, but with a new
+              base symbol named after `pname`, e.g. `aL0(t, x+1, y-2)`, where
+              `aL0` is given by the `pname`.
+        """
+        b = self.base
+        f = self.function
+        strides_map = self.strides_map
+
+        # TODO: resolve circular import. This is a tough one though, as it
```

Comment:
Could you reduce the number of things in `__all__` and then use more specific import paths where necessary?

Reply:
I could, but it wouldn't fix this problem, which is more deeply-rooted than you may think.